File: Exercism Codes/C#/IsbnVerifier.cs
public static class IsbnVerifier
{
    public static bool IsValid(string number)
    {
        string isbn = number.Replace("-", "");
        int n = isbn.Length;
        if (n != 10)
        {
            return false;
        }

        int total = 0;
        for (int i = 0; i < n; i++)
        {
            int m = n - i;
            if (i == n - 1 && isbn[i] == 'X')
            {
                total += 10 * m;
                continue;
            }
            if (char.IsDigit(isbn[i]))
            {
                total += (isbn[i] - '0') * m;
            }
            else
            {
                return false;
            }
        }
        return total % 11 == 0;
    }
}

File: Exercism Codes/Kotlin/IsbnVerifier.kt
class IsbnVerifier {

    fun isValid(number: String): Boolean {
        val isbn10 = number.replace("-", "")
        if (isbn10.length != 10) return false
        var total = 0
        for (idx in 0 until isbn10.length) {
            val n = isbn10.length - idx
            if (idx == isbn10.length - 1 && isbn10[idx] == 'X') {
                total += 10 * n
                continue
            }
            if (isbn10[idx].isDigit()) {
                total += isbn10[idx].toString().toInt() * n
            }
            else {
                return false
            }
        }

        return total % 11 == 0
    }
}
